### PR TITLE
week2 database assignment

### PR DESCRIPTION
**Pull request by daxiousm**

### CODE REVIEW

File: databases/week2/schoolDatabase.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+-- Student: with the columns: id, name, email, phone, class_id (foreign key)
+CREATE TABLE `student` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `name` varchar(255) NOT NULL,
+  `email` varchar(255)  NOT NULL,
+   `phone` varchar(30) NOT NULL,
+    `class_id` int(10) unsigned,
```
Comment (babakbandpey) [inline databases/week2/schoolDatabase.sql:21]: The attribute `class_id `is not belong to the student class.
You need a `class_students `table to save the relations. A class has many students and a student has many classes.

File: databases/week2/schoolDatabase.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    CONSTRAINT `fk_class_id` FOREIGN KEY (`class_id`) REFERENCES `class` (`id`) ON DELETE CASCADE
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+-- Create an index on the name column of the student table.
+
+USE school;
+CREATE INDEX name_index ON student(name);
+
+-- Add a new column to the class table named status which can only have the following values: not-started, ongoing, finished (hint: enumerations).
+ALTER  TABLE class ADD  status ENUM('notstarted', 'ongoing', 'finished') NOT NULL;
```
Comment (babakbandpey) [inline databases/week2/schoolDatabase.sql:32]: The default value must be defined for the new column

File: databases/week2/schoolDatabase.sql [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  `email` varchar(255)  NOT NULL,
+   `phone` varchar(30) NOT NULL,
+    `class_id` int(10) unsigned, 
+   PRIMARY KEY (`id`),
+    CONSTRAINT `fk_class_id` FOREIGN KEY (`class_id`) REFERENCES `class` (`id`) ON DELETE CASCADE
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+-- Create an index on the name column of the student table.
+
+USE school;
```
Comment (babakbandpey) [inline databases/week2/schoolDatabase.sql:28]: Which database is this one?

Comment (daxiousm) [reply databases/week2/schoolDatabase.sql:28]: the schoolDatabase

Comment (babakbandpey) [reply databases/week2/schoolDatabase.sql:28]: You need to correct the PR

File: databases/week2/part4.sql [hunk 4]
```diff
@@ -0,0 +1,40 @@
+
```
[the remark this answers is not in the document]
Comment (daxiousm) [reply databases/week2/part4.sql:1]: i tried creating relationship between singres, albums and songs, please check it out

File: databases/week2/music2.sql [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+-- -----------------------------------------------------
+-- Table `music`.`Albums`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `music`.`Albums` (
+  `AlbumsId` INT NOT NULL AUTO_INCREMENT,
+  `SingersId` INT NOT NULL,
+  `AlbumTitle` VARCHAR(45) NOT NULL,
+  `songs_TrackId` INT NOT NULL,
+  `songs_SingersId` INT NOT NULL,
+  `songs_AlbumsId` INT NOT NULL,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:35]: remove

File: databases/week2/music2.sql [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+-- -----------------------------------------------------
+-- Table `music`.`Albums`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `music`.`Albums` (
+  `AlbumsId` INT NOT NULL AUTO_INCREMENT,
+  `SingersId` INT NOT NULL,
+  `AlbumTitle` VARCHAR(45) NOT NULL,
+  `songs_TrackId` INT NOT NULL,
+  `songs_SingersId` INT NOT NULL,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:34]: remove

File: databases/week2/music2.sql [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+-- Schema music
+-- -----------------------------------------------------
+CREATE SCHEMA IF NOT EXISTS `music` DEFAULT CHARACTER SET utf8 ;
+USE `music` ;
+
+-- -----------------------------------------------------
+-- Table `music`.`songs`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `music`.`songs` (
+  `TrackId` INT NOT NULL AUTO_INCREMENT,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:18]: id

File: databases/week2/music2.sql [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  `SingersId` INT NOT NULL,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:19]: singer_id

File: databases/week2/music2.sql [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+  `AlbumsId` INT NOT NULL,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:20]: album_id

File: databases/week2/music2.sql [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+  `SongName` VARCHAR(45) NULL,
+  PRIMARY KEY (`TrackId`, `SingersId`, `AlbumsId`))
```
Comment (babakbandpey) [inline databases/week2/music2.sql:22]: add singer_id and album_id as foreign keys with reference to their respected tables

File: databases/week2/music2.sql [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `music`.`Albums`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `music`.`Albums` (
+  `AlbumsId` INT NOT NULL AUTO_INCREMENT,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:30]: id

File: databases/week2/music2.sql [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+  `SingersId` INT NOT NULL,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:31]: remove

File: databases/week2/music2.sql [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+  `AlbumTitle` VARCHAR(45) NOT NULL,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:32]: title

File: databases/week2/music2.sql [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+  `songs_TrackId` INT NOT NULL,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:33]: track_id could be in songs table

File: databases/week2/music2.sql [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+  `songs_SingersId` INT NOT NULL,
+  `songs_AlbumsId` INT NOT NULL,
+  PRIMARY KEY (`AlbumsId`, `SingersId`, `AlbumTitle`, `songs_TrackId`, `songs_SingersId`, `songs_AlbumsId`),
```
Comment (babakbandpey) [inline databases/week2/music2.sql:36]: only the id from the table is enough

File: databases/week2/music2.sql [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION)
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `music`.`Singers`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `music`.`Singers` (
+  `SingersId` INT UNSIGNED NOT NULL AUTO_INCREMENT,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:51]: id

File: databases/week2/music2.sql [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+  `FirstName` VARCHAR(45) NULL,
+  `LastName` VARCHAR(45) NULL,
+  `SingerInfo` VARCHAR(45) NULL,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:54]: Bio/info

File: databases/week2/music2.sql [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+  `Albums_AlbumsId` INT NOT NULL,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:55]: remove

File: databases/week2/music2.sql [hunk 19]
```diff
@@ -0,0 +1,94 @@
...
+  `Albums_SingersId` INT NOT NULL,
+  `Albums_AlbumTitle` VARCHAR(45) NOT NULL,
```
Comment (babakbandpey) [inline databases/week2/music2.sql:57]: remove

File: databases/week2/music2.sql [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+  PRIMARY KEY (`SingersId`),
```
Comment (babakbandpey) [inline databases/week2/music2.sql:58]: id

File: databases/week2/music2.sql [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+    FOREIGN KEY (`Albums_AlbumsId` , `Albums_SingersId` , `Albums_AlbumTitle`)
+    REFERENCES `music`.`Albums` (`AlbumsId` , `SingersId` , `AlbumTitle`)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION)
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `music`.`Singers_has_songs`
+-- -----------------------------------------------------
```
Comment (babakbandpey) [inline databases/week2/music2.sql:70]: not needed

File: databases/week2/part1.sql [hunk 22]
```diff
@@ -0,0 +1,27 @@
+INSERT INTO task (title, description, created, updated, due_date, status_id, user_id)
```
Comment (babakbandpey) [inline databases/week2/part1.sql:1]: take a look at this and make it work with the week 2 assignment

File: databases/week2/part1.sql [hunk 23]
```diff
@@ -0,0 +1,27 @@
...
+
+-- Mark a task as complete
+UPDATE task
+SET status_id = 3
+WHERE id = 36;
+
+-- Delete a task
+  DELETE FROM task
+  where id = 36;
+
```
Comment (babakbandpey) [inline databases/week2/part1.sql:27]: Remember to delete from `user_tasks `by `task_id `and then delete from `tasks`

**Review by babakbandpey — CHANGES_REQUESTED** [review]
Hello,

Please make these changes.
Thanks

**Pull request merged**